packages/std/src/ado_contract/execute.rs

```diff
@@ -386,12 +386,26 @@ macro_rules! unwrap_amp_msg {
                     )),
                 }
             );
+
             ctx.deps
                 .api
                 .debug(&format!("Unwrapped msg: {:?}", msg.as_ref()));
             ctx.amp_ctx = Some(pkt);
         }
 
+        if !msg.is_permissionless() {
+            ::cosmwasm_std::ensure!(
+                ::andromeda_std::ado_contract::permissioning::is_context_permissioned(
```

Comment:
nit: Can be done in a single check by doing `msg.is_permissionless() || ...is context permissioned check`